### DISCUSSION TITLE
Ray doesn't work when running tests.

### DISCUSSION THREAD

**Discussion by HassanZahirnia** [spatie/ray#758] [Q&A] [answered]
Hey 👋😄
I got an issue with Ray and thought about asking here.
I run my Laravel projects inside WSL2 (W11) and I can use `ray()` commands pretty much anywhere except when running test units (Pest).

I'm not sure why this is happening. If there is a `ray()` command inside a test file, or like a factory file that would be run by the test, the `ray()` won't send/receive message.

The only way to overcome this is to use `sudo` or login as `root` user. Like:
```
sudo php artisan test
```

Anyone got any ideas ?
Thanks in advance!

**Comment by HassanZahirnia** [comment, marked as answer]
Ok shortly after posting my question, I found out Ray uses curl behind the scenes, and because I happen to have env proxy set as my home user, the tests could not communicate with Ray client....

So basically the solution was:
```
unset HTTP_PROXY
```

I hope this helps someone in the future if they ran into same issue 😊👍

Have a great day and thanks for your great tool! 🌸

**Reply by freekmurze** [reply]
Thanks for letting us know your solution!
